File: android/app/src/main/kotlin/com/sotalog/android/domain/models/QRZCallsignResult.kt
package com.sotalog.android.domain.models

data class QRZCallsignResult(
    val callsign: String,
    val firstName: String? = null,
    val nickname: String? = null,
    val lastName: String? = null,
    val city: String? = null,
    val state: String? = null,
    val country: String? = null,
    val grid: String? = null,
    val county: String? = null,
) {
    val name: String?
        get() {
            if (nickname != null) return nickname
            val full = listOfNotNull(firstName, lastName).joinToString(" ")
            return full.ifEmpty { null }
        }

    val qth: String?
        get() = state ?: country
}

File: ios/SOTALog/Models/QRZCallsignResult.swift
import Foundation

/// Result from QRZ XML callsign lookup
struct QRZCallsignResult: Equatable {
    let callsign: String
    let firstName: String?
    let nickname: String?
    let lastName: String?
    let city: String?
    let state: String?
    let country: String?
    let grid: String?
    let county: String?

    /// Combined name for display
    var name: String? {
        if let nickname { return nickname }
        return [firstName, lastName].compactMap { $0 }.joined(separator: " ").nilIfEmpty
    }

    /// QTH for display — state for US, country otherwise
    var qth: String? {
        state ?? country
    }
}

private extension String {
    var nilIfEmpty: String? {
        isEmpty ? nil : self
    }
}
